vars/WebHookAnalyzer.groovy

```diff
@@ -0,0 +1,264 @@
+ * @param jobName jobName
+ * @param timerConfig cron expression to schedule the job
+ * @return
+ */
+def createJenkinsJob(def jobName, def timerConfig, def file) {
+
+    echo "Creating the job ${jobName}"
+    def jobDSL="@Library('intg_test_template@${LocalProperties.TG_ENV}') _\n" +
+                "Pipeline()"
+    def flowDefinition = new org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition(jobDSL, true)
+    def instance = Jenkins.instance
+    def job = new org.jenkinsci.plugins.workflow.job.WorkflowJob(instance, jobName )
+    job.definition = flowDefinition
+    job.setConcurrentBuild(false)
+
+    if (timerConfig != null && timerConfig != "") {
+        hudson.triggers.TimerTrigger newCron = new hudson.triggers.TimerTrigger(timerConfig);
+        newCron.start(job, true)
+        job.addTrigger(newCron)
+    }
+    def rawYamlLocation = generateRawYamlLocation(file)
+    def prop = new EnvInjectJobPropertyInfo("", "${LocalProperties.TESTGRID_YAML_URL_KEY}=${rawYamlLocation}", "",
+            "", "", false)
+    def prop2 = new org.jenkinsci.plugins.envinject.EnvInjectJobProperty(prop)
+    prop2.setOn(true)
+    prop2.setKeepBuildVariables(true)
+    prop2.setKeepJenkinsSystemVariables(true)
+    job.addProperty(prop2)
+    job.save()
+    Jenkins.instance.reload()
+
+}
+
+String gennerateJobName() {
+    def props = LocalProperties.instance
+    def jobName = props.GIT_REPOSITORY + "-" + props.GIT_BRANCH
+    return jobName
+}
+
+String generateRawYamlLocation(def fileLocation) {
+    // We will split from the repo name and get the rest to create the raw URL
+    def relativePath = fileLocation.split(LocalProperties.GIT_REPOSITORY)[1]
+    return LocalProperties.GH_RAW_URL + "/" + LocalProperties.GIT_ORG_NAME + "/" + LocalProperties.GIT_REPOSITORY +
+            "/" + LocalProperties.GIT_BRANCH + relativePath
+}
+
+/**
+ * Clones a given git repo
+ * @param gitURL
+ * @param gitBranch
+ */
+void cloneRepo(def gitURL, gitBranch) {
+    def props = Properties.instance
+    tryAddKnownHost("github.com")
+    sshagent(credentials: ['github_bot']) {
+        sh """
+            echo Cloning repository: ${gitURL}
+            cd ${props.WORKSPACE}
+            git clone -b ${gitBranch} ${gitURL}
+        """
+    }
+}
+
+/**
+ * Add hostUrl to knownhosts on the system (or container) if necessary so that ssh commands will go
+ * through even if the certificate was not previously seen.
+ * @param hostUrl
+ */
+void tryAddKnownHost(String hostUrl) {
+    // ssh-keygen -F ${hostUrl} will fail (in bash that means status code != 0) if ${hostUrl} is not yet a known host
+    def statusCode = sh script: "ssh-keygen -F ${hostUrl}", returnStatus: true
+    if (statusCode != 0) {
+        sh "mkdir -p ~/.ssh"
+        sh "ssh-keyscan ${hostUrl} >> ~/.ssh/known_hosts"
+    }
+}
+
+def findTestGridYamls(def searchPath) {
+    def files
+    dir(searchPath) {
+        files = findFiles(glob: '**/testgrid.yaml')
```

Comment:
This returns any testgrid.yaml located anywhere within the repo, yes? 
What about the testgrid yamls under the .testgrid/ folder? They won't have the exact testgrid.yaml name.

Reply:
That logic is not implemented yet